elements/pfe-clipboard/pfe-clipboard.ts

```diff
@@ -0,0 +1,323 @@
+    this._checkForCopyTarget();
+
+    // Emit event when this component has connected in case copyContent needs to be set
+    this.dispatchEvent(pfeEvent('connected', { component: this }));
+    /**
+     * @deprecated
+     * @fires {CustomEvent<{ component: PfeClipboard }>} pfe-clipboard:connected
+     */
+    this.dispatchEvent(pfeEvent('pfe-clipboard:connected', { component: this }));
+
+    // This prevents a regression, default text used to be "Copy URL".
+    // Now that component can copy _anything_ that's not ideal default text
+    if (this.copyFrom === 'url' && !this.slots.hasSlotted('text'))
+      this.textSlotDefault = 'Copy URL';
+  }
+
+  render() {
+    return html`
+      <!-- icon slot -->
+      ${!this.noIcon ? html`
+          <div class="pfe-clipboard__icon">
+            <slot name="icon" id="icon">
+              <svg id="icon--url" xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 15.277 16">
+                <g transform="translate(-2.077 -1.807)">
+                  <path class="a" d="M15.34,2.879a3.86,3.86,0,0,0-5.339,0L6.347,6.545a3.769,3.769,0,0,0,0,5.339.81.81,0,0,0,1.132,0,.823.823,0,0,0,0-1.145A2.144,2.144,0,0,1,7.5,7.677l3.641-3.654a2.161,2.161,0,1,1,3.049,3.062l-.8.8a.811.811,0,1,0,1.145,1.132l.8-.8a3.769,3.769,0,0,0,0-5.339Z" transform="translate(0.906 0)"/>
+                  <path class="a" d="M10.482,6.822a.823.823,0,0,0,0,1.145,2.161,2.161,0,0,1,0,3.049L7.343,14.155a2.161,2.161,0,0,1-3.062,0,2.187,2.187,0,0,1,0-3.062l.193-.116a.823.823,0,0,0,0-1.145.811.811,0,0,0-1.132,0l-.193.193a3.86,3.86,0,0,0,0,5.339,3.86,3.86,0,0,0,5.339,0l3.126-3.139A3.731,3.731,0,0,0,12.72,9.562a3.769,3.769,0,0,0-1.094-2.74A.823.823,0,0,0,10.482,6.822Z" transform="translate(0 1.37)"/>
+                </g>
+              </svg>
+              <svg id="icon--copy" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 32 32">
+                <g></g>
+                <path d="M30.286 6.857q0.714 0 1.214 0.5t0.5 1.214v21.714q0 0.714-0.5 1.214t-1.214 0.5h-17.143q-0.714 0-1.214-0.5t-0.5-1.214v-5.143h-9.714q-0.714 0-1.214-0.5t-0.5-1.214v-12q0-0.714 0.357-1.571t0.857-1.357l7.286-7.286q0.5-0.5 1.357-0.857t1.571-0.357h7.429q0.714 0 1.214 0.5t0.5 1.214v5.857q1.214-0.714 2.286-0.714h7.429zM20.571 10.661l-5.339 5.339h5.339v-5.339zM9.143 3.804l-5.339 5.339h5.339v-5.339zM12.643 15.357l5.643-5.643v-7.429h-6.857v7.429q0 0.714-0.5 1.214t-1.214 0.5h-7.429v11.429h9.143v-4.571q0-0.714 0.357-1.571t0.857-1.357zM29.714 29.714v-20.571h-6.857v7.429q0 0.714-0.5 1.214t-1.214 0.5h-7.429v11.429h16z"/>
+              </svg>
+            </slot>
+          </div>
+      ` : ''}
+      <div class="pfe-clipboard__text">
+          <slot name="text" id="text">${this.textSlotDefault}</slot>
+      </div>
+      <div class="pfe-clipboard__text--success" role="alert">
+          <slot name="success-text" id="success-text">Copied</slot>
+          <slot name="text--success" id="text--success">Copied</slot>
+      </div>
```

Comment:
I went for parallel instead of stacked to allow for piecemeal upgrades - meaning so that users can upgrade the dependency on a given page without having to change all their markup at once. but if you think that's not an issue here (and thinking it over, it probably isn't, since it's unlikely that multiple elements would be slotted into this position) then we can stack this one

Reply:
we might also be able to solve by hiding according to `hasSlot`

Reply:
Stacked should still allow for piecemeal upgrades, no?  If there is no element using the `success-text"`slot (new), then it will fallback will expose the `text--succes` slot (old) for elements to opt into.  I tested it locally and it works.

Reply:
please check again

Reply:
yep that works.  The only caveat is that if we go with SSR, It won't slot the content into the shadowRoot until the component hydrates.  But in this case, it won't matter because it needs javascript to show the success message. lol

Reply:
yeah ?hidden might be better, but we can loop back around to that later if we want, maybe with an ssr-compatible directive?

lmk what you think, and plz resolve if you're satisfied

Reply:
Solid plan 👍